rust/alloc/slice.rs

```diff
@@ -472,6 +472,24 @@ impl<T> [T] {
         self.to_vec_in(Global)
     }
 
+    /// Tries to copy `self` into a new `Vec`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let s = [10, 40, 30];
+    /// let x = s.try_to_vec().unwrap();
+    /// // Here, `s` and `x` can be modified independently.
+    /// ```
+    #[inline]
+    #[stable(feature = "kernel", since = "1.0.0")]
```

Comment:
I don't think `stable` is okay here. Some options:
* A custom feature gate, but it is not very easy to use as all crates will have to add them.
* Keep it a `stable`, but gate it under a kernel-specific custom `cfg`. If using this option, `since` still should not be 1.0.0; it needs to be at least the current nightly version.

Reply:
Nobody will use this code except the kernel, i.e. this is not a crate for others to use, and it is in-tree. Thus adding version tracking, feature gates or `cfg` options does not buy us anything, only increases complexity.

Reply:
Well I don't think there are any blockers preventing these changes from being upstreamed under a feature flag.

The main reason I think the current way is not acceptable is that when we generate the docs for the alloc crate, people may look at the doc and think that it is part of standard Rust library but clearly it isn't.

Reply:
> Well I don't think there are any blockers preventing these changes from being upstreamed under a feature flag.

Ah, you meant for upstreaming. For that, things need anyway more changes/steps/work, e.g.
  - Writing more tests/benches in the right places (we don't have all the files here in any case).
  - Adapt and test things in the latest `master`/nightly (for the moment we are pretty close, but we will be ideally jumping from stable to stable release).
  - Group/split the changes in meaningful feature flags as you mention.

So I don't think we can/want to follow the Rust process here. The idea, after all, is to detach ourselves a bit so that we can experiment with the additions and find the right APIs (these additions are straight forward, yes, but perhaps others aren't).

> The main reason I think the current way is not acceptable is that when we generate the docs for the alloc crate, people may look at the doc and think that it is part of standard Rust library but clearly it isn't.

I am unsure why you think that is a big deal -- these are the kernel docs, after all. If somebody gets confused, they will find pretty quickly with a error message when coding for userspace etc.

However, I do think added clarity is good. In fact, that is why I replaced the Rust logo with the Rust for Linux one (in the version we sent for the RFC) so that it was clear when the user is navigating from the kernel docs into the Rust ones (e.g. when clicking on `Some`).

So if it eases your concerns, they will see the logo change at the very least. Having also a banner on the specific methods that are kernel-only would also be nice to further clarify, I will look into that.

Reply:
You can add `#[doc(cfg(kernel))]`

Reply:
Opened https://github.com/Rust-for-Linux/linux/issues/407.